src/com/sun/ts/tests/ejb32/lite/timer/schedule/expire/Client.java

```diff
@@ -449,19 +462,17 @@ public void leapYears() {
         (leapYears[1] - 6) + "-" + leapYears[1], String.valueOf(leapYears[2]) };
     String[] dayOfMonths = { "29, 29", "29, 29", "last, last" };
 
-    Calendar cal = Calendar.getInstance();
+    final Calendar cal = Calendar.getInstance(DateUtils.UTC_TIME_ZONE);
     cal.set(Calendar.DAY_OF_MONTH, 1);
 
     for (int i = 0; i < leapYears.length; i++) {
       ScheduleExpression exp = TimerUtil.getPreciseScheduleExpression(cal)
-          .dayOfMonth(dayOfMonths[i]).month(2).year(yearRange[i]);
-
-      Date expectedNextTimeout = TimerUtil.getCurrentDatePlus(0, 0);
-      expectedNextTimeout = DateUtils.setYears(expectedNextTimeout,
-          leapYears[i]);
-      expectedNextTimeout = DateUtils.setDays(expectedNextTimeout, 29);
-      expectedNextTimeout = DateUtils.setMonths(expectedNextTimeout,
-          Calendar.FEBRUARY);
+          .dayOfMonth(dayOfMonths[i]).month(2).year(yearRange[i]).timezone(DateUtils.UTC_TIME_ZONE.getID());
```

Comment:
It would be better if this and the lines above stuck within the 80 column line width
used by the original code.

Otherwise, looks good to me, but get an EJB expert to look at it too!